### PR TITLE
Improve CI

### PR DESCRIPTION
**Pull request by rhnonose**
- [x] Replace TravisCI with CircleCI
- [x] Fix coveralls
- [x] Add warnings as errors
- [x] Add formatter check
- [x] Add credo check

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+      - restore_cache:
+          keys:
+            - v1-mix-cache-{{ .Branch }}-{{ checksum "mix.lock" }}
+            - v1-mix-cache-{{ .Branch }}
+            - v1-mix-cache
+      - restore_cache:
+          keys:
+            - v1-build-cache-{{ .Branch }}
+            - v1-build-cache
```
Comment (joaodubas) [inline circle.yml]: Why do we have two `restore_cache`  items?

File: circle.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      - save_cache:
+          key: v1-build-cache-{{ .Branch }}
+          paths: "_build"
+      - save_cache:
+          key: v1-build-cache
+          paths: "_build"
+
+      - run:
+          name: Wait for DB
+          command: dockerize -wait tcp://localhost:5432 -timeout 1m
```
Comment (joaodubas) [inline circle.yml]: You could also use the [`pg_isready`][0] command to check connectivity.

[0]: https://www.postgresql.org/docs/10/app-pg-isready.html

File: circle.yml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+      - run: mix test
```
Comment (joaodubas) [inline circle.yml]: We could check the code coverage.

File: circle.yml (hunk 1 above)
Comment (rhnonose) [reply circle.yml]: I got a default template 😂 
What do you recommend?

Comment (joaodubas) [reply circle.yml]: Just thought it was strange, but I can see the reason, we're separating dependency cache from build cache.

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+      - run:
+          name: Wait for DB
+          command: dockerize -wait tcp://localhost:5432 -timeout 1m
+
+      - run: mix coveralls.circle --umbrella
+      - store_artifacts:
+          path: cover/excoveralls.html
+      - store_test_results:
+          path: _build/test/junit
```
Comment (joaodubas) [inline .circleci/config.yml]: Nice 👍

File: circle.yml [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+      - store_test_results:
+          path: /tmp/backend-repor-test-result
+
+      - run: mix coveralls.circle --umbrella
+      - store_artifacts:
+          path: cover/excoveralls.html
+
+      - run: mix credo --strict
```
Comment (joaodubas) [inline circle.yml]: Is this file needed at all?

File: .circleci/config.yml (hunk 4 above)
Comment (rhnonose) [reply .circleci/config.yml]: Not working 😢

File: circle.yml (hunk 5 above)
Comment (rhnonose) [reply circle.yml]: I'll remove after adding stickler

File: .circleci/config.yml (hunk 4 above)
Comment (rhnonose) [reply .circleci/config.yml]: It is now 😂

File: circle.yml (hunk 2 above)
Comment (rhnonose) [reply circle.yml]: Seems unavailable.
Do you think it's worth installing or is it fine this way?

Comment (joaodubas) [reply circle.yml]: It's fine this way!

Comment (joaodubas) [reply circle.yml]: Probably this is the recommendation from circleci.

**Comment by pmargreff** [comment]
Great job :+1: . 

Late feedbacks, coveralls badge doesn't seem to be updating (probably broked link?).

**Comment by rhnonose** [comment]
@pmargreff coveralls was messed up for some reason, looks ok on master
